react/src/AddDatasets/components/UploadDialog.tsx

```diff
@@ -52,7 +54,7 @@ export default function UploadDialog({ open, onClose, groups }: UploadDialogProp
         if (file !== null) {
             // Upload
             const groupsParam = new URLSearchParams({ groups: groups.join(",") });
-            const response = await fetch("/api/_bulk?" + groupsParam.toString(), {
+            const response = await fetch(`${fetchContext}/api/_bulk?` + groupsParam.toString(), {
```

Comment:
I like storing the endpoint in a Context. What do you think about going a step farther and writing a hook that wraps `fetch` itself, either something like `useFetch` (user passes in verb, base endpoint already bound) or `usePost`, `useGet`, etc.? It would take some refactoring of whatever fetch calls aren't currently part of the React ecosystem, but it would save the trouble (now and down the line) of maintaining endpoint configurations for each and every `fetch`.

Reply:
Yeah, that good idea. I'll start looking into that then, & we can hold this PR off for now

Reply:
So, I ended up getting rid of the fetchContext completely. The useFetch accesses the active endpoint from `localStorage`.